### PR TITLE
CI: Fix and simplify the dvc-diff workflow

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Fix the dvc-diff workflow by setting up nodejs to v16, following the workaround in https://github.com/iterative/cml/issues/1377.

The dvc-diff report was posted using `peter-evans/find-comment` and `peter-evans/create-or-update-comment` in the old workflow. These steps are removed and the diff report is posted using a single command `cml comment update report.md`.

Comment https://github.com/GenericMappingTools/pygmt/pull/2549#issuecomment-1566699742 is the comment posted by the old workflow and https://github.com/GenericMappingTools/pygmt/pull/2549#issuecomment-1566763109 is the comment posted by the new workflow (i.e., `cml comment update report.md`).

TODO:

- [ ] Revert changes in b7ad43b724ec46507d49937fb543d27160d50f00 before merging

### CODE REVIEW

**Review by michaelgrund — APPROVED** [review]

File: pygmt/tests/test_basemap.py [hunk 1]
```diff
@@ -11,21 +11,17 @@ def test_basemap():
...
-    Create a loglog basemap plot.
+    Create a simple basemap plot.
     """
     fig = Figure()
-    fig.basemap(
-        region=[1, 10000, 1e20, 1e25],
-        projection="X16cl/12cl",
-        frame=["WS", "x2+lWavelength", "ya1pf3+lPower"],
-    )
+    fig.basemap(region=[10, 70, -20, 20], projection="X10c/6c", frame="afg")
```
Comment (weiji14) [inline pygmt/tests/test_basemap.py:24]: > Revert changes in https://github.com/GenericMappingTools/pygmt/commit/b7ad43b724ec46507d49937fb543d27160d50f00 before merging

Remember to do the revert!

File: .github/workflows/dvc-diff.yml [hunk 2]
```diff
@@ -95,30 +100,5 @@ jobs:
...
-
-    - name: Update comment with new image diff report
-      if: steps.fc.outputs.comment-id != ''
-      uses: peter-evans/create-or-update-comment@v3.0.1
-      with:
-        comment-id: ${{ steps.fc.outputs.comment-id }}
-        body: ${{ steps.image-diff.outputs.report }}
-        edit-mode: replace
+        # create/update PR comment
+        cml comment update report.md
```
Comment (weiji14) [inline .github/workflows/dvc-diff.yml:104]: Nice that this simplifies so many lines of code!

File: .github/workflows/dvc-diff.yml [hunk 3]
```diff
@@ -35,6 +35,12 @@ jobs:
...
 
+    # workaround from https://github.com/iterative/cml/issues/1377
+    - name: Setup NodeJS
+      uses: actions/setup-node@v1
+      with:
+        node-version: '16'
+
     # Produce the markdown diff report, which should look like:
     # ## Summary of changed images
     #
```
Comment (weiji14) [inline .github/workflows/dvc-diff.yml:46]: Could update the preview a little bit below here

```diff
    # | Status   | Path                                |
    # |----------|-------------------------------------|
    # | added    | pygmt/tests/baseline/test_image.png |
    # | deleted  | pygmt/tests/baseline/test_image2.png |
    # | modified | pygmt/tests/baseline/test_image3.png |
+
+    ## Image diff(s)
+
+    <details>
+    ...
+    </details>
+
+    Report last updated at commit abcdef
```

**Review by weiji14 — APPROVED** [review]
Just a small suggestion, but otherwise looks good!

**Pull request merged**